examples/http/helloworld_server/main.go

```diff
@@ -50,7 +50,7 @@ func main() {
 		r, _ := http.NewRequest("GET", "https://example.com", nil)
 
 		// Propagate the trace header info in the outgoing requests.
-		r = req.WithContext(req.Context())
```

Comment:
There definitely looks to be a problem here but I don't think your fix is quite right. `WithContext` doesn't mutate the receiver, it returns a copy of it with the supplied context set. I think this should be:

```go
r = r.WithContext(req.Context())
```

Reply:
Thanks, I have updated PR.